set_user.c

```diff
@@ -149,6 +152,44 @@ extern Datum set_user(PG_FUNCTION_ARGS);
 void _PG_init(void);
 void _PG_fini(void);
 
+static bool
+check_user_whitelist(Oid userId, const char *whitelist)
+{
+	char	   *rawstring;
+	List	   *userlist;
+	ListCell   *l;
+
+	if (whitelist == NULL || whitelist[0] == '\0')
+		return false;
+
+	rawstring = pstrdup(whitelist);
+
+	/* Parse string into list of identifiers */
+	if (!SplitIdentifierString(rawstring, ',', &userlist))
+	{
+		/* syntax error in list */
+		pfree(rawstring);
```

Comment:
Freeing memory here is unnecessary...

Why is this LOG and not ERROR? Assuming the latter, the return is also unnecessary.